app.py

```diff
@@ -46,5 +55,37 @@ def subtype_detail(story_type_name, subtype_name):
     return render_template('subtype_detail.html', story_type=story_type, subtype=subtype)
 
 
+@app.route('/subtype/<story_type_name>/<subtype_name>/update', methods=['POST'])
+def update_story_selection(story_type_name, subtype_name):
+    """Handle form submission for story element selections."""
+    story_type = registry.get_story_type(story_type_name)
```

Comment:
You should be employing the story class here.   i.e. the core_arc and key_theme selections should be in the story class